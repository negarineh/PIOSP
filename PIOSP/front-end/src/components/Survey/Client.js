/* eslint-disable no-undef */
function search(cb) {
    return fetch(`save`, {
      accept: "application/json",
      method: "GET"
    })
      .then(checkStatus)
      .then(parseJSON)
      .then(cb)
      .catch(err => console.log(err));
  }
  
  function checkStatus(response) {
    if (response.status >= 200 && response.status < 300) {
      return response;
    }
    const error = new Error(`HTTP Error ${response.statusText}`);
    error.status = response.statusText;
    error.response = response;
    console.log(error); // eslint-disable-line no-console
    throw error;
  }
  
  function parseJSON(response) {
    console.log(response.json);
    return response.json();
  }
  
  const Client = { search };
  export default Client;