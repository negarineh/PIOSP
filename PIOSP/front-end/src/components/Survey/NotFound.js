import React from 'react';

export default class NotFound extends React.Component {

    render(){
        return(
            <div className="page-not-found">
                <h4>Page Not Found</h4>
            </div>
        );
    }
}