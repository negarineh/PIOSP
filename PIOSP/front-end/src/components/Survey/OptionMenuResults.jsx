import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../stylesheets/MainPage.css';
import {AdminList} from '../../AdminListPage';
import {NumberOfAnswersCategoryChart} from '../../SurveyResultsPage';
import {NumberOfCorrectAnswersCategory} from '../../SurveyResultsPage';
import {NumberOfCorrectAnswers} from '../../SurveyResultsPage';

import { Segment, Header, Dropdown } from 'semantic-ui-react';
import ErrorBoundry from '../../ErrorBoundry/ErrorBoundry';

    const tagOptionsMenu = [
      {
        text: 'Answers Based on Category',
        value: 'Answers Based on Category',
        // label: { color: 'red', empty: true, circular: true },
        icon: {name:'table', color:'black'}
      },
      {
        text: 'Correct Answers Based on Category',
        value: 'Correct Answers Based on Category',
        // label: { color: 'black', empty: true, circular: true,  },
        icon: {name:'table', color:'black'},
      },
      {
        text: 'Number of Correct and InCorrect Answers',
        value: 'Number of Correct and InCorrect Answers',
        // label: { color: 'black', empty: true, circular: true,  },
        icon: {name:'table', color:'black'},
      },
    ];


class OptionMenuResults extends Component {


  constructor(props) {
    super(props);
    this.state = {
      activeItem: 'home',
      visible: false,
      option:'',
    };
  }
  toggleVisibility = () => this.setState({ visible: !this.state.visible })

  handleItemClick = (e, { name }) => {
    this.setState({ activeItem: name });
  }


  handleChange = (e, { name, value }) => 
  {
    this.setState({ [name]: value , option: value});
    if (value === 'Profile')
        alert("Profile");
    if (value === 'SignOut')
        alert('SignOut');
    if (value === 'Table 1')
        return (<AdminList/>);
  }

  render() {

    const { option } = this.state;

    return (
      <div >
      
        <Segment style={{height: '638.5px'}}>
          <Header as='h2' textAlign='left'>
              Choose Graph
          </Header>
            <Dropdown
                selection
                name='default'
                options={tagOptionsMenu}
                placeholder='Choose the Graph'
                onChange={this.handleChange.bind(this)}
                onClick={this.handleItemClick.bind(this)}
                fluid
              />
              {option ==='Answers Based on Category'?
                <ErrorBoundry>
                  <NumberOfAnswersCategoryChart/>
                </ErrorBoundry>:
                  null }
              {option ==='Correct Answers Based on Category'?
                <ErrorBoundry>
                  <NumberOfCorrectAnswersCategory/>
                </ErrorBoundry>:
                  null }
              {option ==='Number of Correct and InCorrect Answers'?
                <ErrorBoundry>
                  <NumberOfCorrectAnswers/>
                </ErrorBoundry>:
                null }
        </Segment>
      </div>
    );
  }
} 

export default OptionMenuResults;

OptionMenuResults.defaultProps = {
  visibel: false,
  toggleVisibility: () => {},
  handleItemClick: () => {},
  handleChange: () => {}
}

// prop checks
OptionMenuResults.propTypes = {
  visible: PropTypes.bool.isRequired,
  toggleVisibility: PropTypes.func.isRequired,
  handleItemClick: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
}
