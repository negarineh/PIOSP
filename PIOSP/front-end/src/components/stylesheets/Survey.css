.form-control {
  margin-bottom: 5px;
  margin-right: 5px; }

.btn-primary2s {
  position: relative;
  display: block;
  margin: 30px auto 0;
  padding: 10px 30px;
  font-size: 15px;
  font-family: sans-serif;
  font-weight: 200;
  color: white;
  background-image: linear-gradient(#38cf38, #21ca0a);
  border: none;
  z-index: 1;
  border-radius: 5px; }
  .btn-primary2s:before {
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 0;
    height: 100%;
    background-image: linear-gradient(#0a830a, #21ca0a);
    transition: width 250ms;
    z-index: -1;
    border-radius: 5px; }
  .btn-primary2s:hover {
    color: white;
    border-radius: 5px; }
    .btn-primary2s:hover:before {
      width: 100%; }

.labelpage1 {
  color: #463831;
  font-family: sans-serif;
  font-size: 15px; }

.psurvey1 {
  float: left;
  align-items: left;
  height: 10%;
  color: #463831;
  padding-bottom: 5px;
  padding-right: 30%;
  line-height: 1.5;
  font-family: sans-serif;
  font-size: 15px; }

.psurvey2 {
  float: left;
  height: 10%;
  width: 80%;
  color: #463831;
  padding-bottom: 5pX;
  padding-top: 10px;
  line-height: 1.5;
  font-family: sans-serif;
  font-size: 15px; }

.psurvey21 {
  height: 10%;
  width: 100%;
  color: #463831;
  padding-bottom: 5pX;
  padding-top: 10px;
  line-height: 1.5;
  font-family: sans-serif;
  font-size: 15px;
  text-align: center; }

.fieldsetsurvey {
  text-align: left;
  padding: 20px;
  padding-bottom: 20px;
  margin-top: 20px;
  margin-left: 20px;
  margin-right: 20px;
  margin-bottom: 20px;
  border-radius: 10px; }

.legendsurvey {
  font-family: sans-serif;
  font-size: 15px; }

.imgsurvey {
  width: 480px;
  height: 430px;
  clear: none;
  border-style: double;
  border-color: #485f19;
  border-radius: 10px; }

.question {
  float: left;
  width: 50%;
  padding-left: 5%; }

.form-control-description-s {
  width: 420px;
  height: 80px;
  font-family: "fantasy";
  font-size: 20px;
  border-radius: 10px;
  resize: none; }

.surveygroup1 {
  float: left;
  width: 50%; }

.form-group-description {
  float: left; }

.wrapperWelcome {
  text-align: center;
  align-content: center;
  align-items: center;
  justify-content: center;
  align-self: center; }

.pwelcome {
  align-items: center;
  color: #463831;
  padding-bottom: 20px;
  line-height: 1.5;
  font-family: sans-serif;
  font-size: 15px;
  text-align: center;
  position: relative; }

.btn-primary2100 {
  position: relative;
  margin-left: 10px;
  padding: 10px 30px;
  font-size: 20px;
  font-family: "fantasy";
  font-weight: 400;
  color: white;
  background-image: linear-gradient(#20d31a, #8ac208);
  border: none;
  z-index: 1;
  border-radius: 5px; }
  .btn-primary2100:before {
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 0;
    height: 100%;
    background-image: linear-gradient(#0a830a, #21ca0a);
    transition: width 250ms;
    z-index: -1;
    border-radius: 5px; }
  .btn-primary2100:hover {
    color: white;
    border-radius: 5px; }
    .btn-primary2100:hover:before {
      width: 100%; }
