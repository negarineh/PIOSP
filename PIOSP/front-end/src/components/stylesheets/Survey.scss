 // Variables
 $color-bg: white;
 $color-main: #333;
 $color-hover: lighten(#0275d8, 1%);
 $padding: 20px;
 $radius: 10px;
 $accent-color: #dfcfad; 
 $font-size:20px;
 $font-family: 'fantasy';
 $background-color: rgb(78, 82, 76);
 $background-color-after: #e74c3c;
 
 // Mixins
 @mixin transition($prop: all, $duration: 0.3s, $timing: ease-in-out) {
   transition: $prop $duration $timing;
 }
  
 
 // h1 {
 //   color: $color-main;
 //   font-size: 30px;
 //   margin-top: 0;
 //   text-align: center; 
 //   font-family: 'fantasy';
 // }
  
 .form-control { 
 //   border: none;
 //   border-bottom: 1px solid $color-main;
 //   background: none;
 //   width: 100%;
 //   font-size: 18px;
 //   font-weight: bold;
 //   line-height: 1.2;
 //   outline: none;
 //   text-align: right;
 //   border-radius: $radius;
 //   vertical-align: bottom;
 //   position: relative;
 //   // float: left;
 //   // *overflow: hidden;
     margin-bottom: 5px;
     margin-right: 5px;
 }
 // .btn:hover {
 //   background: $color-hover;
 //   border-color: $color-hover;
 // }
    
 // .wrapperpage1 {      
 //   margin: 50px;
 //   font-size: 20px;
 //   // text-align: center;
 //   width: 1200px; 
 //   height: 550px;
 //   border-radius: $radius;
 //   background: transparent;
 //   box-shadow: 0 15px 35px rgba(28, 28, 204, 0.1), 0 5px 15px rgba(0, 0, 0, 0.514);
 //   // padding: $padding;
 //   background-color: white;
 //   }  
 
 .btn-primary2s{
             position: relative;
             display: block;
             margin: 30px auto 0;
             padding: 10px 30px;
             font-size: 15px;
             font-family: sans-serif;
             font-weight: 200;
             color: white;
             background-image: linear-gradient(#38cf38,#21ca0a);
             border: none;
             z-index: 1;
             border-radius: 5px;
             &:before {
                 position: absolute;
                 content: '';
                 top: 0;
                 left: 0;
                 width: 0;
                 height: 100%;
                 background-image: linear-gradient(#0a830a,#21ca0a);
                 transition: width 250ms;
                 z-index: -1;
                 border-radius: 5px;
             }
             
             &:hover {
                 color: white;
                 border-radius: 5px;
                 
                 &:before {
                     width: 100%;
                 }
             }
   }
 
 .labelpage1{
   color: rgb(70, 56, 49);
   // background: $accent-color;
   font-family: sans-serif;
  font-size: 15px; 
 //   display: block; 
//  float: left;
//   clear: none;
 }
 .psurvey1{
  //  display: flex;
  // clear: none;
  float: left;
  align-items: left ;
 //   justify-content: center;
   // position: absolute;
   height: 10%; 
  //  width: 80%;
   color: rgb(70, 56, 49);
   padding-bottom: 5px;;
   padding-right: 30%;
   line-height: 1.5;
   font-family: sans-serif;
  font-size: 15px;
 }
  
 .psurvey2{
  //  display: flex;
  // clear: none;
  float: left;
  //  align-items: left ;
 //   justify-content: center;
   // position: absolute;
   height: 10%; 
   width: 80%;
   color: rgb(70, 56, 49);
   padding-bottom: 5pX;
  padding-top: 10px;
  line-height: 1.5;
  font-family: sans-serif;
  font-size: 15px;
 }

 .psurvey21{
  //  display: flex;
  // clear: none;
  // float: left;
  //  align-items: left ;
 //   justify-content: center;
   // position: absolute;
   height: 10%; 
   width: 100%;
   color: rgb(70, 56, 49);
   padding-bottom: 5pX;
  padding-top: 10px;
  line-height: 1.5;
  font-family: sans-serif;
  font-size: 15px;
  text-align: center;
 }
 
 .fieldsetsurvey{
   text-align: left;
   padding: $padding; 
   padding-bottom: 20px; 
   margin-top: $padding;  
   margin-left: $padding;
   margin-right: $padding;
   margin-bottom: $padding;
   border-radius: $radius; 
 }
 
 .legendsurvey{
     // padding-top: 10px;
     font-family: sans-serif;
     font-size: 15px;
     
 }
 
 // .wrapper1.input{
 //   align-self: auto;
 //   float: left;
 // }
 // .wrapper1.label{
 //   text-align: right;
 //   float: left;
 // }
 
 // .form-group{
 //   padding: 5px 5px 5px 5px;
 //   left: 60px;
 //   position: absolute;
 // }
 
 .imgsurvey{
    width:480px;
    height:430px;
    // float: left;
    clear: none;
    // display: inline-block;
    border-style: double;
    border-color: rgb(72, 95, 25);
    border-radius: $radius;
  }

 .question{
  //  display: inline-block;
   float: left;
   width: 50%;
  //  padding-right: 70%;
   padding-left: 5%;
 }

 .form-control-description-s{ 
  width:420px;
  height:80px;
  font-family: "fantasy";
  font-size: 20px;
  border-radius: $radius;
  resize: none;
} 

// .body-component-a{
//   background-color: #e74c3c;
// }

.surveygroup1{
  float: left; 
  width: 50%;
}

.form-group-description{
  float: left;
}

.wrapperWelcome { 
  text-align: center;
  align-content: center;
  align-items: center;
  justify-content: center;
  align-self: center;}

  .pwelcome {
    // display: flex;
    align-items: center;
    // height: 100%;
    // width: 100%;
    color: #463831;
    padding-bottom: 20px;
    line-height: 1.5;
    font-family: sans-serif;
    font-size: 15px; 
    text-align: center;
    position: relative;}

    .btn-primary2100 {
      position: relative;
            //  display: block;
            //  margin: 30px auto 0;
            margin-left: 10px;
             padding: 10px 30px;
             font-size: $font-size;
             font-family: $font-family;
             font-weight: 400;
             color: white;
             background-image: linear-gradient(#20d31a, #8ac208);
             border: none;
             z-index: 1;
             border-radius: 5px;
             &:before {
                 position: absolute;
                 content: '';
                 top: 0;
                 left: 0;
                 width: 0;
                 height: 100%;
                 background-image: linear-gradient(#0a830a,#21ca0a);
                 transition: width 250ms;
                 z-index: -1;
                 border-radius: 5px;
             }
             
             &:hover {
                 color: white;
                 border-radius: 5px;
                 
                 &:before {
                     width: 100%;
                 }
             }
    }
    