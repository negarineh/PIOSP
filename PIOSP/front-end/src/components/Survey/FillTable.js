import React from 'react';
import PropTypes from 'prop-types';

import '../stylesheets/Survey.css';
import '../stylesheets/MainPage.css';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Segment, Message} from 'semantic-ui-react';
import '../stylesheets/Survey.css';

var globalResponse = [];

// filling the pollinator table when we are running the server for the first time
class FillTable extends React.Component {
    constructor(props) {
      super(props);   
      this.state={
        photo:[],
        response: [],
        response1: [],
        APIResponse:'',
      };
      this.handleSubmit = this.handleSubmit.bind(this);
    }
  
    componentDidMount() {

      fetch('http://localhost:3000/checkFill')
              .then(data => data.json())
              .then((data) => { globalResponse = data;
              console.log(globalResponse); // eslint-disable-line no-console
            }).catch(err => console.error(err));
    }
    
    handleSubmit(){

      fetch('http://localhost:3000/checkFill')
              .then(data => data.json())
              .then((data) => { globalResponse = data;
              console.log(globalResponse); // eslint-disable-line no-console
            }).catch(err => console.log(err));

      if (globalResponse.length !== 0)
        console.log('Filled'); // eslint-disable-line no-console
      else if (globalResponse.length === 0)
      {
        fetch('http://localhost:3000/fill')
              .then(datas => datas.json())
              .then((datas) => { this.setState({ photo: datas });
            }).catch(err => console.log(err));
      }  
    }

    render() {

      return (
        /* jshint ignore:start */
    <div>
      <MuiThemeProvider>
            <Segment style={{height: '638.5px'}}>
                <Message
                    positive
                    header='Filling Photo Table'
                    content="First before running the survey you should fill the photo table with names of all photos."
                    style={{textAlign:'left'}}
                    />
                      <button 
                          type="submit" 
                          className="btn btn-primary2100"
                          onClick={this.handleSubmit}
                          >
                            Fill Pollinators Table
                      </button>
            </Segment>
      </MuiThemeProvider>
    </div>
    /* jshint ignore:end */
      );
    }
  }

FillTable.defaultProps = {
    dispatch: () => {},
  };
  
// prop checks
FillTable.propTypes = {
  // users: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default FillTable;