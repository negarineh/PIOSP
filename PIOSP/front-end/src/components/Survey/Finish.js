import React from 'react';
import PropTypes from 'prop-types';

import '../stylesheets/Survey.css';
import '../stylesheets/MainPage.css';
import '../stylesheets/Survey.css';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Segment, Step} from 'semantic-ui-react';
import {Link} from 'react-router-dom';


// last page of survey 
class Finish extends React.Component {
    constructor(props) {
      super(props);   
      this.state={
        stepIndex: 0,
      };
    }
    
  render() {
 
  return (
    <div className="mainpage">
     <MuiThemeProvider>
      {/* <Header/> */}
      <div className="wrappermainpage">      

      <div style={{width: 'auto', margin: 'auto', color:"dimgray", paddingLeft:"10px", overflow:'auto'}}>
      <Step.Group ordered size="mini" >
              <Step completed >
                <Step.Content>
                  <Step.Title>Enrolling</Step.Title>
                  <Step.Description>Insert your email </Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Confirmed Email </Step.Title>
                  <Step.Description>Welcome to our Survey</Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Survey Questions</Step.Title>
                  <Step.Description>Please answer questions</Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Rating and Further Experiences</Step.Title>
                  <Step.Description>Please rate our survey</Step.Description>
                </Step.Content>
              </Step>

              <Step active >
                <Step.Content>
                  <Step.Title>Survey Submitted</Step.Title>
                  <Step.Description>Thanks for your participation</Step.Description>
                </Step.Content>
              </Step>
            </Step.Group>
        </div>
        <Segment style={{margin:"10px", overflow:"auto"}}>
            <div style={{textAlign:'center'}}>
              <p key="a" className="psurvey21" > 
              Thank for your participation
              </p>
              
              <button 
                          type="submit" 
                          className="btn btn-primary2100"
                          
                          >
                  {/* Survey Home Page */}
                <Link  to={"/mainpage"}>Survey Home Page</Link>
              </button>
            </div>
        </Segment>
      </div>
      </MuiThemeProvider>
    </div>
      );
    }
  }

  Finish.defaultProps = {
    history: {},
  }

// prop checks
  Finish.propTypes = {
    history: PropTypes.object.isRequired,
};

export default Finish;