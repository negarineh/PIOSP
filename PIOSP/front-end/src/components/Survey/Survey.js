import React from 'react';
import PropTypes from 'prop-types';

import '../stylesheets/Survey.css';
import '../stylesheets/MainPage.css';
import Client from './Client';
import { Button, Segment, Label, Step} from 'semantic-ui-react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import cookie from 'react-cookies';

class Survey extends React.Component {
    constructor(props) {
      super(props);   
      this.state={
        AppSource: '',
        description: '',
        photo:'',
        photos: [],
        items: [],
        sendItems: props.sendItems,
        userId: '',
        number: 0,
      };
      this.addItem = this.addItem.bind(this);
    }

    componentDidMount() {
    
    Client.search(photos => {
      this.setState({
        photos: photos,
        number: 0
      });
    });
    // console.log(this.state.photos);
    
    this.setState ({ userId: cookie.load('jwt') });

    window.setTimeout(function(){

    // Move to a new location or you can do something else
    window.location.href = "http://localhost:3001/submit";
    cookie.remove('jwt', { path: '/' });
    }, 15 * 60 * 1000);

    }

    addItem(e) {

      e.preventDefault();
      const data = new FormData(e.target);
      
      var curAppSource = data.get('AppSource'),
          curDescription = data.get('description'),
          curPhoto = data.get('photo'),
          curUserId = cookie.load('jwt');

      var itemArray = this.state.items;
          // sendItems = this.state.items;
      var alwaysTrue = true;
  // try {
      if (alwaysTrue) {
        // itemArray.unshift({
        //     description: this._inputElement.value,
        //     photo: this.state.photos[this.state.number].slice(6, this.state.photos[this.state.number].length-4),
        //     answer: this.AppSource.value ,
        //     id: this.state.userId
        // });
  
              // On submit of the form, send a POST request with the data to the server.
            fetch(`save`,{
              method: 'POST',
              body:JSON.stringify({ 
                AppSource: curAppSource,
                description: curDescription,
                photo: curPhoto,
                id: curUserId
              }),
              headers: {"Content-Type": "application/json"}
            })
            .then(function(response){
              if (response.status >= 200 && response.status < 300) {
              return response;
              }
              const error = new Error(`HTTP Error ${response.statusText}`);
              error.status = response.statusText;
              error.response = response;
              console.log(error);
              throw error;
          }, function(err){console.error(err);});

        this.setState({
          items: itemArray,
          sendItems: itemArray,
          number: this.state.number + 1
        });
      }
    // } catch (err){
    //   console.log(err);
    // }
      // this.setState({
        
      // });
      
      this._inputElement.value = "";
      this.AppSource.checked = false;

      console.log(itemArray); // eslint-disable-line no-console
      console.log(this.state.number); // eslint-disable-line no-console
      // e.preventDefault();

      if (this.state.number === 14){
      // console.log(sendItems[0].answer); // eslint-disable-line no-console

      this.props.history.push("/submit");
    }

      
      console.log(this.state.number); // eslint-disable-line no-console

    }
 
    render() {

      let AppSource = this.state.AppSource;

      const { photos } = this.state;

      return (
    <div className="mainpage">
      <MuiThemeProvider>
      {!AppSource &&
        <div className="wrappermainpage" >      
            <div style={{width: 'auto', margin: 'auto', color:"dimgray", paddingLeft:"10px", overflow:'auto'}}>
            <Step.Group ordered size="mini" >
              <Step completed >
                <Step.Content>
                  <Step.Title>Enrolling</Step.Title>
                  <Step.Description>Insert your email </Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Confirmed Email </Step.Title>
                  <Step.Description>Welcome to our Survey</Step.Description>
                </Step.Content>
              </Step>

              <Step active >
                <Step.Content>
                  <Step.Title>Survey Questions</Step.Title>
                  <Step.Description>Please answer questions</Step.Description>
                </Step.Content>
              </Step>

              <Step disabled >
                <Step.Content>
                  <Step.Title>Rating and Further Experiences</Step.Title>
                  <Step.Description>Please rate our survey</Step.Description>
                </Step.Content>
              </Step>

              <Step disabled >
                <Step.Content>
                  <Step.Title>Survey Submitted</Step.Title>
                  <Step.Description>Thanks for your participation</Step.Description>
                </Step.Content>
              </Step>
            </Step.Group>
            </div>
        
        <Segment style={{margin:"10px", overflow:"auto"}}>
            <Label as='a' image><img src={"/"+ photos[this.state.number]} alt=""/>
            Image {this.state.number+1} of 15</Label>
            <form  onSubmit={this.addItem}> 
            <div className="surveygroup1">
            <p key="a" className="psurvey1" > The insect in the image is:(You can select one only)
            </p>
              
              <div key="b" className="question">
                <div className="form-group">
                 <label className="labelpage1">
                  <input 
                     type="radio" 
                     name="AppSource" 
                     className="form-control"
                     value="Bee"
                     ref={(a) => this.AppSource = a}
                     required />
                     Bee
                      </label>
                </div>
                <div className="form-group">
                 <label className="labelpage1">
                  <input 
                     type="radio" 
                     name="AppSource" 
                     className="form-control"
                     value="Beetle"
                     ref={(a) => this.AppSource = a}
                     required />
                     Beetle
                      </label>
                </div>
                <div className="form-group">
                 <label className="labelpage1">
                  <input 
                     type="radio" 
                     name="AppSource" 
                     className="form-control"
                     value="Fly"
                     ref={(a) => this.AppSource = a}
                     required />
                     Fly
                      </label>
                </div>
                <div className="form-group">
                 <label className="labelpage1">
                  <input 
                    //  placeholder= "Email"
                     type="radio" 
                     name="AppSource" 
                     className="form-control"
                     value="Moth/Butterfly"
                     ref={(a) => this.AppSource = a}
                     required />
                     Moth/Butterfly
                      </label>
                </div>
                <div className="form-group">
                 <label className="labelpage1">
                  <input 
                     type="radio" 
                     name="AppSource" 
                     className="form-control"
                     value="Wasp"
                     ref={(a) => this.AppSource = a}
                     required />
                     Wasp
                      </label>
                </div>
                <div className="form-group">
                 <label className="labelpage1">
                  <input 
                     type="radio" 
                     name="AppSource" 
                     className="form-control"
                     value="Don't know"
                     ref={(a) => this.AppSource = a}
                     required />
                     I don't know
                      </label>
                </div>
              </div>
              <p key="d" className="psurvey2"  > (Optional) If you can further identify this insect, please do so in the box below(e.g.Group, genus, species, common name):
                </p>
            {/* A JSX comment */}

                <div key="e" className="form-group-description">
                  <textarea 
                     type="input" 
                     name="description" 
                     className="form-control-description-s"
                     ref={(a) => this._inputElement = a}
                     />
                </div>
                <Button content='Save and Next' icon='right arrow' labelPosition='right' className="btn-primary2s" style={{width:"200px", height:"50px", backgroundImage: "linear-gradient(#20d31a, #8ac208)" , marginTop:"15px", paddingTop:"5px", color:"white", textShadow:"0 -1px 0 #7c7a17", fontFamily:  "sans-serif", fontSize:'15px', fontWeight:"100"}}/>
              </div>

                <div  key="c" className="photossurvey">
                    <img name="photo" className="imgsurvey" src={"/"+ photos[this.state.number]}  alt=" "/>
                    
                </div>
              </form>
              </Segment>
          </div>
  }
          </MuiThemeProvider>
          </div>
      );
    }
  }

  Survey.defaultProps = {
    sendItems: {},
    addItem: () => {},
    history: {},
  }

  Survey.propTypes = {
    sendItems: PropTypes.object.isRequired,
    addItem: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired
};

export default Survey;