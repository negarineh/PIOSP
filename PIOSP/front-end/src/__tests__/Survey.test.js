import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import Survey from "../components/Survey/Survey";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('Survey component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {sendItems: {}, history: {}, addItem: ()=>{}};
      const wrapper = shallow(<Survey {...props}/>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
