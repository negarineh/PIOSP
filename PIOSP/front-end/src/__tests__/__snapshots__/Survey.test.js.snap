// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey component render() should render the component 1`] = `
<div
  className="mainpage"
>
  <MuiThemeProvider>
    <div
      className="wrappermainpage"
    >
      <div
        style={
          Object {
            "color": "dimgray",
            "margin": "auto",
            "overflow": "auto",
            "paddingLeft": "10px",
            "width": "auto",
          }
        }
      >
        <StepGroup
          ordered={true}
          size="mini"
        >
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Enrolling
              </StepTitle>
              <StepDescription>
                Insert your email 
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Confirmed Email 
              </StepTitle>
              <StepDescription>
                Welcome to our Survey
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            active={true}
          >
            <StepContent>
              <StepTitle>
                Survey Questions
              </StepTitle>
              <StepDescription>
                Please answer questions
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            disabled={true}
          >
            <StepContent>
              <StepTitle>
                Rating and Further Experiences
              </StepTitle>
              <StepDescription>
                Please rate our survey
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            disabled={true}
          >
            <StepContent>
              <StepTitle>
                Survey Submitted
              </StepTitle>
              <StepDescription>
                Thanks for your participation
              </StepDescription>
            </StepContent>
          </Step>
        </StepGroup>
      </div>
      <Segment
        style={
          Object {
            "margin": "10px",
            "overflow": "auto",
          }
        }
      >
        <Label
          as="a"
          image={true}
        >
          <img
            alt=""
            src="/undefined"
          />
          Image 
          1
           of 15
        </Label>
        <form
          onSubmit={[Function]}
        >
          <div
            className="surveygroup1"
          >
            <p
              className="psurvey1"
              key="a"
            >
               The insect in the image is:(You can select one only)
            </p>
            <div
              className="question"
              key="b"
            >
              <div
                className="form-group"
              >
                <label
                  className="labelpage1"
                >
                  <input
                    className="form-control"
                    name="AppSource"
                    required={true}
                    type="radio"
                    value="Bee"
                  />
                  Bee
                </label>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="labelpage1"
                >
                  <input
                    className="form-control"
                    name="AppSource"
                    required={true}
                    type="radio"
                    value="Beetle"
                  />
                  Beetle
                </label>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="labelpage1"
                >
                  <input
                    className="form-control"
                    name="AppSource"
                    required={true}
                    type="radio"
                    value="Fly"
                  />
                  Fly
                </label>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="labelpage1"
                >
                  <input
                    className="form-control"
                    name="AppSource"
                    required={true}
                    type="radio"
                    value="Moth/Butterfly"
                  />
                  Moth/Butterfly
                </label>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="labelpage1"
                >
                  <input
                    className="form-control"
                    name="AppSource"
                    required={true}
                    type="radio"
                    value="Wasp"
                  />
                  Wasp
                </label>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="labelpage1"
                >
                  <input
                    className="form-control"
                    name="AppSource"
                    required={true}
                    type="radio"
                    value="Don't know"
                  />
                  I don't know
                </label>
              </div>
            </div>
            <p
              className="psurvey2"
              key="d"
            >
               (Optional) If you can further identify this insect, please do so in the box below(e.g.Group, genus, species, common name):
            </p>
            <div
              className="form-group-description"
              key="e"
            >
              <textarea
                className="form-control-description-s"
                name="description"
                type="input"
              />
            </div>
            <Button
              as="button"
              className="btn-primary2s"
              content="Save and Next"
              icon="right arrow"
              labelPosition="right"
              style={
                Object {
                  "backgroundImage": "linear-gradient(#20d31a, #8ac208)",
                  "color": "white",
                  "fontFamily": "sans-serif",
                  "fontSize": "15px",
                  "fontWeight": "100",
                  "height": "50px",
                  "marginTop": "15px",
                  "paddingTop": "5px",
                  "textShadow": "0 -1px 0 #7c7a17",
                  "width": "200px",
                }
              }
            />
          </div>
          <div
            className="photossurvey"
            key="c"
          >
            <img
              alt=" "
              className="imgsurvey"
              name="photo"
              src="/undefined"
            />
          </div>
        </form>
      </Segment>
    </div>
  </MuiThemeProvider>
</div>
`;
